lib/palindrome_permutation.rb

```diff
@@ -1,4 +1,26 @@
 
 def palindrome_permutation?(string)
+  #takes a string as an argument and returns true if the letters could be re-arranged into a palindrome  
+  hash = {}
   
+  string.split("").each do |char|
+    if hash[char]
+      hash[char] += 1
+    else
+      hash[char] = 1
+    end
+  end
+  
+  found_an_odd_value = nil
+  hash.values.each do |value|
+    if value % 2 == 1
+      if found_an_odd_value
+        return false
+      else
+        found_an_odd_value = true
+      end
+    end
+  end
```

Comment:
The logic here looks correct to me but the nested `if` statements make it a little harder to follow. One way to avoid this may be to use a counter instead of a boolean for keeping track of previously found odd values:

```
odd_value_count = 0
hash.values.each do |value|
  odd_value_count += 1 if value % 2 == 1
  return false if odd_value_count > 1
end
```